lib/mais_orcid_client.rb

```diff
@@ -0,0 +1,119 @@
+# frozen_string_literal: true
+
+require "active_support/core_ext/module/delegation"
+require "active_support/core_ext/hash/indifferent_access"
+
+require "faraday"
+require "faraday/retry"
+require "oauth2"
+require "singleton"
+require "zeitwerk"
+
+# Load the gem's internal dependencies: use Zeitwerk instead of needing to manually require classes
+Zeitwerk::Loader.for_gem.setup
+
+# Client for interacting with MAIS's ORCID API
+class MaisOrcidClient
+  include Singleton
+
+  # Struct for ORCID-relating data returned from API.
+  OrcidUser = Struct.new(:sunetid, :orcidid, :scope, :access_token, :last_updated) do
+    def update?
+      scope.include?("/activities/update")
+    end
```

Comment:
Didn't know about adding private methods to Structs, cool. I don't see this `update?` method in the code anywhere so far--is the idea that it will be needed in subsequent work? Or did I miss something?

Reply:
It is used in sul_pub code.